lib/pengine/rules.c

```diff
@@ -1032,7 +1032,11 @@ pe__eval_attr_expr(xmlNodePtr expr, pe_rule_eval_data_t *rule_data)
     if (value != NULL && h_val != NULL) {
         if (type == NULL) {
             if (pcmk__str_any_of(op, "lt", "lte", "gt", "gte", NULL)) {
-                type = "number";
+                if (pcmk__char_in_any_str('.', h_val, value, NULL)) {
+                    type = "number";
+                } else {
+                    type = "integer";
+                }
```

Comment:
I think we concluded that backward compatibility should not be an issue, because integers were represented as 32-bit ints and could thus be represented precisely as doubles.

So, if a user has configured type="number" explicitly and they were using numbers that could be represented as 32-bit ints, then representing those numbers as doubles instead (due to explicit "number") should produce the same comparison results.

The same cannot be said for the new long long integer representations after this commit. But any rule expressions that rely on long long above int's capacity would have to be new; there can't be any pre-existing ones.

It could break if the user was previously comparing two unequal values, each greater than or equal to INT_MAX, and expecting them to be equal. But that would be considered a bug in the first place.

Reply:
Oh right, I forgot. It's good as is.